presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/LogicalCteOptimizer.java

```diff
@@ -319,9 +325,10 @@ public Optional<String> peekActiveCte()
 
         public void addDependency(String currentCte)
         {
-            graph.addNode(currentCte);
+            cteDependencyGraph.addNode(currentCte);
             Optional<String> parentCte = peekActiveCte();
-            parentCte.ifPresent(s -> graph.putEdge(currentCte, s));
+            // (current -> parentCte) this indicates that currentCte must be processed first
```

Comment:
Fixed wrong comments about the dependency graph and added explanations

Reply:
I could reverse the graph to be according to the comments and then add a reverse in topological ordering, but then i would have to update some tests (since topological ordering is not unique). Its easier to update the comments